eZ/Publish/Core/Persistence/Tests/DatabaseConnectionFactory.php

```diff
@@ -52,6 +61,10 @@ public function __construct(iterable $databasePlatforms, EventManager $eventMana
      */
     public function createConnection(string $databaseURL): Connection
     {
+        if (isset(self::$connectionPool[$databaseURL])) {
```

Comment:
This is useful for SQLite in-memory tests where connection needs to be kept to avoid loosing data. However storage-based db tests can also benefit from this when it comes to performance.